### PR TITLE
added delimiter checker

### PR DESCRIPTION
**Pull request by bhushankhope**
- Added code to verify the delimiter used in the input file, and to read the file accordingly based on the identified delimiter.
- Added the current change to the changelog.md file

### CODE REVIEW

File: process.py [hunk 1]
```diff
@@ -451,7 +458,7 @@ def process_arguments()->Tuple[TextIO, TextIO, bool, bool, bool]:
     print_tree = args.verbose
     only_one_parent = args.unique
     missing_feature_ok = args.missing
-    return in_file, out_file, dot_file, print_tree, only_one_parent, missing_feature_ok
+    return in_file, out_file, dot_file, print_tree, only_one_parent, missing_feature_ok,file_reader
```
Comment (bherr2) [inline process.py]: Please be mindful of formatting. Need a space after the last comma.

File: process.py [hunk 2]
```diff
@@ -498,9 +505,8 @@ def process_arguments()->Tuple[TextIO, TextIO, bool, bool, bool]:
     max_references: int = 0
 
     INPUT_FILE, OUTPUT_FILE, DOT_FILE, PRINT_TREE, SINGLE_PARENT,\
-    IS_MISSING_FEATURE = process_arguments()
+    IS_MISSING_FEATURE,CONTENTS = process_arguments()
```
Comment (bherr2) [inline process.py]: Please be mindful of formatting. Need a space after the last comma.

File: process.py (hunk 1 above)
Comment (bhushankhope) [reply process.py]: Fixed formatting

File: process.py (hunk 2 above)
Comment (bhushankhope) [reply process.py]: Fixed formatting

**Review by bherr2 — APPROVED** [review]

**Pull request merged**